staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -92,21 +92,26 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 			return
 		}
 
-		patchJS, err := readBody(req)
+		patchBytes, err := readBody(req)
 		if err != nil {
 			scope.err(err, w, req)
 			return
 		}
 
 		ae := request.AuditEventFrom(ctx)
-		audit.LogRequestPatch(ae, patchJS)
+		audit.LogRequestPatch(ae, patchBytes)
 		trace.Step("Recorded the audit event")
 
-		s, ok := runtime.SerializerInfoForMediaType(scope.Serializer.SupportedMediaTypes(), runtime.ContentTypeJSON)
-		if !ok {
-			scope.err(fmt.Errorf("no serializer defined for JSON"), w, req)
+		baseContentType := runtime.ContentTypeJSON
+		if patchType == types.ApplyPatchType {
```

Comment:
This feels like the sort of thing that should be implemented in the patcher type. I think it's OK for now, if this grows more if statements we'll want to move it to keep it readable/testable.